var/spack/repos/builtin.mock/packages/test-build-callbacks/package.py

```diff
@@ -2,23 +2,22 @@
 # Spack Project Developers. See the top-level COPYRIGHT file for details.
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
-
 from spack.package import *
-from spack.package_base import run_after
 
 
-class TestBuildCallbacks(Package):
+class TestBuildCallbacks(MakefilePackage):
     """This package illustrates build callback test failure."""
 
     homepage = "http://www.example.com/test-build-callbacks"
     url = "http://www.test-failure.test/test-build-callbacks-1.0.tar.gz"
 
     version("1.0", "0123456789abcdef0123456789abcdef")
 
-    phases = ["build", "install"]
     # Include undefined method (runtime failure) and 'test' (audit failure)
     build_time_test_callbacks = ["undefined-build-test", "test"]
-    run_after("build")(Package._run_default_build_time_test_callbacks)
+
+    def edit(self, spec, prefix):
+        pass
```

Comment:
This was also before 9981a5d4e127512cf00126bc0939cb1d292e1b56 I can try to revert the change if you think it's better. The aim was to avoid custom phases.

Reply:
That is my preference.

Reply:
Ok, there is also another thing. `Package._run_default_build_time_test_callbacks` changed signature, and now accepts a builder. It should probably be moved outside of `Package`. I'll check how to re-organize that.

Reply:
This package currently is used to test 3 different things: a package audit, using build-time callbacks and a use of `spack test`. The current state is the best way to avoid adding more boilerplate and continue passing all 3 tests. We can either keep this is in a list for later TODO, or I can code additional mock packages so this specific package is used only in a single context.

Reply:
Reworked and extracted a new mock package `fail-test-audit` for 2 of the unit-tests above